### PR TITLE
libxx: All cxx libraries move down one level

### PR DESCRIPTION
**Pull request by W-M-R**
*Note: Please adhere to [Contributing Guidelines](https://github.com/apache/nuttx/blob/master/CONTRIBUTING.md).*

## Summary
Since there are some patches that need to be loaded manually, the original method is rather confusing. Now these patches that need to be loaded manually are classified into various cxx library folders, and each dynamically downloaded cxx library is moved down one directory to make the directory structure clearer.
## Impact
No
## Testing
No

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



No. This PR does not adequately meet the NuttX requirements.  While it provides a brief summary, it lacks crucial details.  Here's why:

* **Insufficient Summary:**  "All cxx libraries move down one level" is vague. *Why* was this move necessary? What problem does it solve?  What specific libraries are affected? What does "down one level" mean in terms of directory structure?
* **Missing Issue References:** Are there related NuttX or NuttX Apps issues that prompted this change? These should be linked.
* **Impact Underexplored:**  Simply stating "No" for all impact categories is insufficient. Even seemingly simple changes can have unforeseen consequences.  Consider build processes, dependencies, and potential compatibility issues. Justify *why* there's no impact in each area. For example, if there's no impact on the build, explain why moving the libraries doesn't affect build scripts or dependencies.
* **Inadequate Testing:**  Stating "depends on another PR" isn't enough. While dependencies exist, this PR needs its *own* verification. Provide specific build host and target details, along with "before" and "after" logs demonstrating the change's effectiveness and the lack of unintended side effects.  What happens if the other PR isn't merged?  How is this PR tested independently?

**In short, the PR needs to be significantly more detailed and self-contained to meet the NuttX requirements.**  It should clearly explain the motivation, implementation details, potential impact, and provide independent verification of its functionality.

**Review by acassis — APPROVED** [review]
Approved, but please include an explanatory message in the commit log as to why this was necessary?

**Comment by W-M-R** [comment]
> Approved, but please include an explanatory message in the commit log as to why this was necessary?

OKey

File: libs/libxx/uClibc++/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# ##############################################################################
+# libs/libxx/uClibc++.cmake
```
Comment (xiaoxiang781216) [inline libs/libxx/uClibc++/CMakeLists.txt:2]:
```suggestion
# llibs/libxx/uClibc++/CMakeLists.txt

```

File: libs/libxx/uClibc++/Make.defs [hunk 2]
```diff
@@ -23,27 +23,29 @@
 UCLIBCXX_VERSION=0.2.5
 
 # Download and unpack tarball if no git repo found
-ifeq ($(wildcard uClibc++/.git),)
```
Comment (xiaoxiang781216) [inline libs/libxx/uClibc++/Make.defs:26]: change line 1 to:
```
libs/libxx/uClibc++/Make.defs
```

File: libs/libxx/libcxxabi/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,125 @@
+# ##############################################################################
+# libs/libxx/libcxxabi.cmake
```
Comment (xiaoxiang781216) [inline libs/libxx/libcxxabi/CMakeLists.txt:2]:
```suggestion
# libs/libxx/libcxxabi/CMakeLists.txt
```

File: libs/libxx/libcxx/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,127 @@
+# ##############################################################################
+# libs/libxx/libcxx.cmake
```
Comment (xiaoxiang781216) [inline libs/libxx/libcxx/CMakeLists.txt:2]:
```suggestion
# libs/libxx/libcxx/CMakeLists.txt
```

File: libs/libxx/uClibc++/CMakeLists.txt (hunk 1 above)
Comment (xiaoxiang781216) [reply libs/libxx/uClibc++/CMakeLists.txt:2]: please update ALL files

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by lupyuen** [comment]
Sorry I have to cancel the CI Check, there's a problem with CI Test:
- https://github.com/apache/nuttx/pull/14667#issuecomment-2461197521

**Comment by W-M-R** [comment]
> Sorry I have to cancel the CI Check, there's a problem with CI Test:
> 
> * [drivers/pipes: return after short write if buffer is full #14667 (comment)](https://github.com/apache/nuttx/pull/14667#issuecomment-2461197521)

Do we need to restart ci after the fix is ​​merged?

**Comment by lupyuen** [comment]
@W-M-R Hopefully not? Let us track down the root cause and figure out the impact. Thanks!

**Comment by lupyuen** [comment]
Sorry I cancelled the CI Check again, since there's a problem with CI Test:
- https://github.com/apache/nuttx/pull/14667#issuecomment-2461197521

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**